### ISSUE TITLE
Invalid road classification for link roads

### ISSUE THREAD

**Issue by jothirnadh** [osmlab/osmlint#248]
Recently came across some link roads which are tagged with wrong road classification. I think it's possible to write linter for this cases and detect all the mistagged roads. 

**Example:**
<img width="655" alt="screen shot 2017-08-08 at 11 02 34 am" src="https://user-images.githubusercontent.com/8401827/29057850-c0c70060-7c2a-11e7-99e9-a5d1333c6d28.png">

This is not a priority. I would like to take this as a learning experience and learn how to write this linter.

cc:// @Rub21

**Comment by planemad** [comment]
A valid condition for a `motorway_link` is that one of its end is connected to another segment of `motorway_link` or `motorway`. This could be extended to any class of road as well.

**Comment by jothirnadh** [comment]
Yep, this is the logic I am thinking as well @planemad, every link road should be connected to the same class of road at least at one end. If this condition is not satisfied it should be detected by this linter.
